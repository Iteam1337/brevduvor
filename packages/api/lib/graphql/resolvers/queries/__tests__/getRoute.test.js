const { ApolloServer } = require('apollo-server-express')
const { createTestClient } = require('apollo-server-testing')
const { gql } = require('apollo-server-express')
const { getRoute } = require('./../index')
const { serverConfig, osrmInstance } = require('./../../../../../index')
const { query, mutate } = createTestClient(new ApolloServer(serverConfig))

// Mock api call
osrmInstance.getTrip = jest.fn(() => {
  return require('./getRouteFixture')
})

describe('getRoute', () => {
  it('exists', () => {
    expect(getRoute).toBeDefined()
  })

  it('returns a Route', async () => {
    const GET_ROUTE = gql`
      query($start: DestinationInput!, $stop: DestinationInput!) {
        getRoute(start: $start, stop: $stop) {
          trips {
            geoJson {
              coordinates
              type
            }
          }
        }
      }
    `

    const res = await query({
      query: GET_ROUTE,
      variables: {
        start: { alias: 'Storuman', lat: 65.090833, lon: 17.1075 },
        stop: { alias: 'Slussfors', lat: 65.4308046, lon: 16.2481741 },
      },
    })

    expect(res.data).toMatchSnapshot()
  })
})
