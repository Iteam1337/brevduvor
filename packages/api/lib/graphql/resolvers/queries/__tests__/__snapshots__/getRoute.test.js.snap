// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRoute fetches a Route with list of coordinates 1`] = `
Object {
  "getRoute": Object {
    "trips": Array [
      Object {
        "geoJson": Object {
          "coordinates": Array [
            Array [
              17.107784,
              65.090853,
            ],
            Array [
              17.123965,
              65.099161,
            ],
            Array [
              17.120946,
              65.114733,
            ],
            Array [
              17.108782,
              65.124776,
            ],
            Array [
              17.103349,
              65.138783,
            ],
            Array [
              17.076498,
              65.147254,
            ],
            Array [
              17.030761,
              65.168542,
            ],
            Array [
              17.018613,
              65.179706,
            ],
            Array [
              17.020304,
              65.189801,
            ],
            Array [
              16.996644,
              65.20245,
            ],
            Array [
              16.987009,
              65.214105,
            ],
            Array [
              16.973307,
              65.221406,
            ],
            Array [
              16.905891,
              65.243612,
            ],
            Array [
              16.858481,
              65.249726,
            ],
            Array [
              16.838497,
              65.255996,
            ],
            Array [
              16.828954,
              65.266083,
            ],
            Array [
              16.810109,
              65.277208,
            ],
            Array [
              16.808482,
              65.284324,
            ],
            Array [
              16.799028,
              65.291041,
            ],
            Array [
              16.768453,
              65.302534,
            ],
            Array [
              16.723265,
              65.303543,
            ],
            Array [
              16.66006,
              65.32586,
            ],
            Array [
              16.632997,
              65.327938,
            ],
            Array [
              16.588506,
              65.352569,
            ],
            Array [
              16.533458,
              65.361462,
            ],
            Array [
              16.520056,
              65.36821,
            ],
            Array [
              16.509276,
              65.382056,
            ],
            Array [
              16.488253,
              65.384175,
            ],
            Array [
              16.440026,
              65.397262,
            ],
            Array [
              16.393008,
              65.396446,
            ],
            Array [
              16.353689,
              65.40526,
            ],
            Array [
              16.332314,
              65.414834,
            ],
            Array [
              16.319168,
              65.4153,
            ],
            Array [
              16.307072,
              65.419247,
            ],
            Array [
              16.291341,
              65.4192,
            ],
            Array [
              16.274764,
              65.429375,
            ],
            Array [
              16.247902,
              65.430578,
            ],
            Array [
              16.274764,
              65.429375,
            ],
            Array [
              16.291341,
              65.4192,
            ],
            Array [
              16.307072,
              65.419247,
            ],
            Array [
              16.319168,
              65.4153,
            ],
            Array [
              16.332314,
              65.414834,
            ],
            Array [
              16.353689,
              65.40526,
            ],
            Array [
              16.393008,
              65.396446,
            ],
            Array [
              16.440026,
              65.397262,
            ],
            Array [
              16.488253,
              65.384175,
            ],
            Array [
              16.509276,
              65.382056,
            ],
            Array [
              16.520056,
              65.36821,
            ],
            Array [
              16.533458,
              65.361462,
            ],
            Array [
              16.588506,
              65.352569,
            ],
            Array [
              16.632997,
              65.327938,
            ],
            Array [
              16.66006,
              65.32586,
            ],
            Array [
              16.723265,
              65.303543,
            ],
            Array [
              16.768453,
              65.302534,
            ],
            Array [
              16.799028,
              65.291041,
            ],
            Array [
              16.808482,
              65.284324,
            ],
            Array [
              16.810109,
              65.277208,
            ],
            Array [
              16.828954,
              65.266083,
            ],
            Array [
              16.838497,
              65.255996,
            ],
            Array [
              16.858481,
              65.249726,
            ],
            Array [
              16.905891,
              65.243612,
            ],
            Array [
              16.973307,
              65.221406,
            ],
            Array [
              16.987009,
              65.214105,
            ],
            Array [
              16.996644,
              65.20245,
            ],
            Array [
              17.020304,
              65.189801,
            ],
            Array [
              17.018613,
              65.179706,
            ],
            Array [
              17.030761,
              65.168542,
            ],
            Array [
              17.076498,
              65.147254,
            ],
            Array [
              17.103349,
              65.138783,
            ],
            Array [
              17.108782,
              65.124776,
            ],
            Array [
              17.120946,
              65.114733,
            ],
            Array [
              17.123965,
              65.099161,
            ],
            Array [
              17.107784,
              65.090853,
            ],
          ],
        },
      },
    ],
  },
}
`;
